vms/platformvm/txs/executor/tx_mempool_verifier.go

```diff
@@ -123,20 +124,30 @@ func (v *MempoolTxVerifier) standardBaseState() (state.Diff, error) {
 	return state, nil
 }
 
-func (v *MempoolTxVerifier) nextBlockTime(state state.Diff) (time.Time, error) {
+func NextBlockTime(state state.Chain, clk *mockable.Clock) (time.Time, bool, error) {
 	var (
-		parentTime  = state.GetTimestamp()
-		nextBlkTime = v.Clk.Time()
+		timestamp  = clk.Time()
+		parentTime = state.GetTimestamp()
 	)
-	if parentTime.After(nextBlkTime) {
-		nextBlkTime = parentTime
+	if parentTime.After(timestamp) {
+		timestamp = parentTime
 	}
+	// [timestamp] = max(now, parentTime)
+
 	nextStakerChangeTime, err := GetNextStakerChangeTime(state)
 	if err != nil {
-		return time.Time{}, fmt.Errorf("could not calculate next staker change time: %w", err)
+		return time.Time{}, false, fmt.Errorf("failed getting next staker change time: %w", err)
 	}
-	if !nextBlkTime.Before(nextStakerChangeTime) {
-		nextBlkTime = nextStakerChangeTime
+
+	// timeWasCapped means that [timestamp] was reduced to
+	// [nextStakerChangeTime]. It is used as a flag for [buildApricotBlock] to
+	// be willing to issue an advanceTimeTx. It is also used as a flag for
+	// [buildBanffBlock] to force the issuance of an empty block to advance
```

Comment:
`buildBanffBlock` doesn't exist in this package - so I'm not sure this comment makes sense.